t/plugin/splunk-hec-logging.t

```diff
@@ -0,0 +1,259 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+
+add_block_preprocessor(sub {
+    my ($block) = @_;
+
+    if ((!defined $block->error_log) && (!defined $block->no_error_log)) {
+        $block->set_value("no_error_log", "[error]");
+    }
+
+    if (!defined $block->request) {
+        $block->set_value("request", "GET /t");
+    }
+
+});
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: full configuration verification
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.splunk-hec-logging")
+            local ok, err = plugin.check_schema({
+                endpoint = {
+                    uri = "http://127.0.0.1:18088/services/collector",
+                    token = "BD274822-96AA-4DA6-90EC-18940FB2414C",
+                    channel = "FE0ECFAD-13D5-401B-847D-77833BD77131",
+                    timeout = 60
+                },
+                max_retry_count = 0,
+                retry_delay = 1,
+                buffer_duration = 60,
+                inactive_timeout = 2,
+                batch_max_size = 10,
+            })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 2: minimize configuration verification
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.splunk-hec-logging")
+            local ok, err = plugin.check_schema({
+                endpoint = {
+                    uri = "http://127.0.0.1:18088/services/collector",
+                    token = "BD274822-96AA-4DA6-90EC-18940FB2414C",
+                }
+            })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: undefined uri configuration
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.splunk-hec-logging")
+            local ok, err = plugin.check_schema({
+                endpoint = {
+                    token = "BD274822-96AA-4DA6-90EC-18940FB2414C",
+                }
+            })
+
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("passed")
+            end
+        }
+    }
+--- response_body
+property "endpoint" validation failed: property "uri" is required
+
+
+
+=== TEST 4: undefined token configuration
+--- config
+    location /t {
+        content_by_lua_block {
+            local plugin = require("apisix.plugins.splunk-hec-logging")
+            local ok, err = plugin.check_schema({
```

Comment:
We can use table drive test for check_schema. You can take this as an example: https://github.com/apache/apisix/blob/c178435d7ada4eeb713d9a1688fb5f54f971abdf/t/plugin/gzip.t#L494